openfe/protocols/openmm_utils/charge_generation.py

```diff
@@ -408,12 +414,105 @@ def assign_offmol_partial_charges(
         generate_n_conformers=generate_n_conformers,
     )  # type: ignore
 
-    # Call selected method to assign partial charges
-    CHARGE_METHODS[method.lower()]['charge_func'](
-        offmol=offmol_copy,
-        toolkit_registry=toolkits,
-        **CHARGE_METHODS[method.lower()]['charge_extra_kwargs'],
-    )  # type: ignore
+    # limit the number of threads used by SQM
+    # <https://github.com/openforcefield/openff-toolkit/issues/1831>
+    with threadpool_limits(limits=2):
```

Comment:
What's the advantage of 2 vs 1 here?

Reply:
I thought 1 might have been harsh but it seems fine in testing, maybe slightly faster while using more cores so changed to 1.